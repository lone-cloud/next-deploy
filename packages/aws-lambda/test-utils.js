const fse = require('fs-extra');
const path = require('path');
const os = require('os');
const LambdaComponent = require('./serverless');

const createTmpDir = () => {
  return fse.mkdtemp(path.join(os.tmpdir(), 'test-'));
};

const createComponent = async () => {
  // create tmp folder to avoid state collisions between tests
  const tmpFolder = await createTmpDir();

  const component = new LambdaComponent('TestLambda', {
    stateRoot: tmpFolder,
  });

  await component.init();

  return component;
};

module.exports = { createComponent, createTmpDir };
