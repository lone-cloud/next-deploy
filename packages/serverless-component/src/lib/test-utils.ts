import fse from 'fs-extra';
import path from 'path';

import { BUILD_DIR } from '../constants';

export const cleanupFixtureDirectory = (fixtureDir: string) => (): Promise<void> => {
  return fse.remove(path.join(fixtureDir, BUILD_DIR));
};
