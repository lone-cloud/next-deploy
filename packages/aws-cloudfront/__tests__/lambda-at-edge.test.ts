import { createComponent, assertHasCacheBehavior } from './test-utils';

import { mockCreateDistribution, mockCreateDistributionPromise } from 'aws-sdk';
import CloudFrontComponent from '../serverless';

jest.mock('aws-sdk', () => require('../__mocks__/aws-sdk.mock'));

describe('Input origin as a custom url', () => {
  let component: CloudFrontComponent;

  beforeEach(async () => {
    mockCreateDistributionPromise.mockResolvedValueOnce({
      Distribution: {
        Id: 'distribution123',
      },
    });

    component = await createComponent();
  });

  it('creates distribution with lambda associations for each event type', async () => {
    await component.default({
      origins: [
        {
          url: 'https://exampleorigin.com',
          pathPatterns: {
            '/some/path': {
              ttl: 10,
              'lambda@edge': {
                'viewer-request': 'arn:aws:lambda:us-east-1:123:function:viewerRequestFunction',
                'origin-request': 'arn:aws:lambda:us-east-1:123:function:originRequestFunction',
                'origin-response': 'arn:aws:lambda:us-east-1:123:function:originResponseFunction',
                'viewer-response': 'arn:aws:lambda:us-east-1:123:function:viewerResponseFunction',
              },
            },
          },
        },
      ],
    });

    assertHasCacheBehavior(mockCreateDistribution, {
      PathPattern: '/some/path',
      LambdaFunctionAssociations: {
        Quantity: 4,
        Items: [
          {
            EventType: 'viewer-request',
            LambdaFunctionARN: 'arn:aws:lambda:us-east-1:123:function:viewerRequestFunction',
            IncludeBody: true,
          },
          {
            EventType: 'origin-request',
            LambdaFunctionARN: 'arn:aws:lambda:us-east-1:123:function:originRequestFunction',
            IncludeBody: true,
          },
          {
            EventType: 'origin-response',
            LambdaFunctionARN: 'arn:aws:lambda:us-east-1:123:function:originResponseFunction',
            IncludeBody: true,
          },
          {
            EventType: 'viewer-response',
            LambdaFunctionARN: 'arn:aws:lambda:us-east-1:123:function:viewerResponseFunction',
            IncludeBody: true,
          },
        ],
      },
    });

    expect(mockCreateDistribution.mock.calls[0][0]).toMatchSnapshot();
  });

  it('throws error when event type provided is not valid', async () => {
    expect.assertions(1);

    try {
      await component.default({
        origins: [
          {
            url: 'https://exampleorigin.com',
            pathPatterns: {
              '/some/path': {
                ttl: 10,
                'lambda@edge': {
                  'invalid-eventtype':
                    'arn:aws:lambda:us-east-1:123:function:viewerRequestFunction',
                },
              },
            },
          },
        ],
      });
    } catch (err) {
      expect(err.message).toEqual(
        '"invalid-eventtype" is not a valid lambda trigger. See https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/lambda-cloudfront-trigger-events.html for valid event types.'
      );
    }
  });
});
