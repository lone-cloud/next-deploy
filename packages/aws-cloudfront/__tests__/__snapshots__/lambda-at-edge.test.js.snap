// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input origin as a custom url creates distribution with lambda associations for each event type 1`] = `
Object {
  "DistributionConfig": Object {
    "Aliases": Object {
      "Items": Array [],
      "Quantity": 0,
    },
    "CacheBehaviors": Object {
      "Items": Array [
        Object {
          "AllowedMethods": Object {
            "CachedMethods": Object {
              "Items": Array [
                "GET",
                "HEAD",
              ],
              "Quantity": 2,
            },
            "Items": Array [
              "GET",
              "HEAD",
            ],
            "Quantity": 2,
          },
          "Compress": true,
          "DefaultTTL": 10,
          "FieldLevelEncryptionId": "",
          "ForwardedValues": Object {
            "Cookies": Object {
              "Forward": "all",
            },
            "Headers": Object {
              "Items": Array [],
              "Quantity": 0,
            },
            "QueryString": true,
            "QueryStringCacheKeys": Object {
              "Items": Array [],
              "Quantity": 0,
            },
          },
          "LambdaFunctionAssociations": Object {
            "Items": Array [
              Object {
                "EventType": "viewer-request",
                "IncludeBody": true,
                "LambdaFunctionARN": "arn:aws:lambda:us-east-1:123:function:viewerRequestFunction",
              },
              Object {
                "EventType": "origin-request",
                "IncludeBody": true,
                "LambdaFunctionARN": "arn:aws:lambda:us-east-1:123:function:originRequestFunction",
              },
              Object {
                "EventType": "origin-response",
                "IncludeBody": true,
                "LambdaFunctionARN": "arn:aws:lambda:us-east-1:123:function:originResponseFunction",
              },
              Object {
                "EventType": "viewer-response",
                "IncludeBody": true,
                "LambdaFunctionARN": "arn:aws:lambda:us-east-1:123:function:viewerResponseFunction",
              },
            ],
            "Quantity": 4,
          },
          "MaxTTL": 10,
          "MinTTL": 10,
          "PathPattern": "/some/path",
          "SmoothStreaming": false,
          "TargetOriginId": "exampleorigin.com",
          "TrustedSigners": Object {
            "Enabled": false,
            "Quantity": 0,
          },
          "ViewerProtocolPolicy": "https-only",
        },
      ],
      "Quantity": 1,
    },
    "CallerReference": "1566599541192",
    "Comment": "",
    "DefaultCacheBehavior": Object {
      "AllowedMethods": Object {
        "CachedMethods": Object {
          "Items": Array [
            "HEAD",
            "GET",
          ],
          "Quantity": 2,
        },
        "Items": Array [
          "HEAD",
          "GET",
        ],
        "Quantity": 2,
      },
      "Compress": false,
      "DefaultTTL": 86400,
      "FieldLevelEncryptionId": "",
      "ForwardedValues": Object {
        "Cookies": Object {
          "Forward": "none",
        },
        "Headers": Object {
          "Items": Array [],
          "Quantity": 0,
        },
        "QueryString": false,
        "QueryStringCacheKeys": Object {
          "Items": Array [],
          "Quantity": 0,
        },
      },
      "LambdaFunctionAssociations": Object {
        "Items": Array [],
        "Quantity": 0,
      },
      "MaxTTL": 31536000,
      "MinTTL": 0,
      "SmoothStreaming": false,
      "TargetOriginId": "exampleorigin.com",
      "TrustedSigners": Object {
        "Enabled": false,
        "Items": Array [],
        "Quantity": 0,
      },
      "ViewerProtocolPolicy": "redirect-to-https",
    },
    "Enabled": true,
    "HttpVersion": "http2",
    "Origins": Object {
      "Items": Array [
        Object {
          "CustomHeaders": Object {
            "Items": Array [],
            "Quantity": 0,
          },
          "CustomOriginConfig": Object {
            "HTTPPort": 80,
            "HTTPSPort": 443,
            "OriginKeepaliveTimeout": 5,
            "OriginProtocolPolicy": "https-only",
            "OriginReadTimeout": 30,
            "OriginSslProtocols": Object {
              "Items": Array [
                "TLSv1.2",
              ],
              "Quantity": 1,
            },
          },
          "DomainName": "exampleorigin.com",
          "Id": "exampleorigin.com",
          "OriginPath": "",
        },
      ],
      "Quantity": 1,
    },
    "PriceClass": "PriceClass_All",
  },
}
`;
